roles/matrix-bot-maubot/templates/config/config.yaml.j2

```
@@ -0,0 +1,129 @@
+    upload: /data/plugins
+    # The directories from which plugins should be loaded.
+    # Duplicate plugin IDs will be moved to the trash.
+    load:
+    - /data/plugins
+    trash: /data/trash
+
+# Configuration for storing plugin databases
+plugin_databases:
+    # The directory where SQLite plugin databases should be stored.
+    sqlite: /data/dbs
+    # The connection URL for plugin databases. If null, all plugins will get SQLite databases.
+    # If set, plugins using the new asyncpg interface will get a Postgres connection instead.
+    # Plugins using the legacy SQLAlchemy interface will always get a SQLite connection.
+    #
+    # To use the same connection pool as the default database, set to "default"
+    # (the default database above must be postgres to do this).
+    #
+    # When enabled, maubot will create separate Postgres schemas in the database for each plugin.
+    # To view schemas in psql, use `\dn`. To view enter and interact with a specific schema,
+    # use `SET search_path = name` (where `name` is the name found with `\dn`) and then use normal
+    # SQL queries/psql commands.
+    postgres:
+    # Maximum number of connections per plugin instance.
+    postgres_max_conns_per_plugin: 3
+    # Overrides for the default database_opts when using a non-"default" postgres connection string.
+    postgres_opts: {}
+
+server:
+    # The IP and port to listen to.
+    hostname: 0.0.0.0
+    port: 29316
+    # Public base URL where the server is visible.
+    public_url: {{ matrix_bot_maubot_bot_server }}
+    # The base management API path.
+    base_path: /_matrix/maubot/v1
+    # The base path for the UI.
+    ui_base_path: /_matrix/maubot
+    # The base path for plugin endpoints. The instance ID will be appended directly.
+    plugin_base_path: /_matrix/maubot/plugin/
+    # Override path from where to load UI resources.
+    # Set to false to using pkg_resources to find the path.
+    override_resource_path: /opt/maubot/frontend
+    # The base appservice API path. Use / for legacy appservice API and /_matrix/app/v1 for v1.
+    appservice_base_path: /_matrix/app/v1
+    # The shared secret to sign API access tokens.
+    # Set to "generate" to generate and save a new token at startup.
+    unshared_secret: {{ matrix_bot_maubot_secret }}
+
+# Known homeservers. This is required for the `mbc auth` command and also allows
+# more convenient access from the management UI. This is not required to create
+# clients in the management UI, since you can also just type the homeserver URL
+# into the box there.
+homeservers:
+  {{  matrix_domain }}:
+    # Client-server API URL
+        url: {{ matrix_bot_maubot_bot_server }}
+    # registration_shared_secret from synapse config
+    # You can leave this empty if you don't have access to the homeserver.
+    # When this is empty, `mbc auth --register` won't work, but `mbc auth` (login) will.
+        secret: {{ matrix_registration_shared_secret }}
```

```suggestion
        secret: {{ matrix_bot_maubot_registration_shared_secret|to_json }}
```

This does a couple of things:
- it adds `|to_json`
- **and more importantly**, to switches away from using the `matrix_registration_shared_secret` variable which is **part of the `matrix-registration` role**.

You probably intended to use `matrix_synapse_registration_shared_secret` (in the case of Synapse, at least).

In any case, the shared secret (coming from the Synapse role.. or Dendrite role) should be passed via another variable (let's call it `matrix_bot_maubot_registration_shared_secret`), which is defined as empty in `roles/matrix-bot-maubot/defaults/main.yml` and is later redefined in terms of `matrix_synapse_registration_shared_secret` (or the Dendrite equivalent) via `group_vars/matrix_servers`.

`group_vars/matrix_servers`:

```yaml
matrix_bot_maubot_registration_shared_secret: |-
  {{
    {
      'synapse': matrix_synapse_registration_shared_secret,
      'dendrite': matrix_dendrite_registration_shared_secret,
    }[matrix_homeserver_implementation]
  }}
```

---

Done